hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/ContainerEndpoint.java

```diff
@@ -170,11 +170,18 @@ public Response getKeysForContainer(
       for (ContainerKeyPrefix containerKeyPrefix : containerKeyPrefixMap
           .keySet()) {
 
-        // Directly calling get() on the Key table instead of iterating since
-        // only full keys are supported now. When we change to using a prefix
-        // of the key, this needs to change to prefix seek.
-        OmKeyInfo omKeyInfo = omMetadataManager.getKeyTable(getBucketLayout())
+        // Directly calling getSkipCache() on the Key/FileTable table
+        // instead of iterating since only full keys are supported now. We will
+        // try to get the OmKeyInfo object by searching the KEY_TABLE table with
+        // the key prefix. If it's not found, we will then search the FILE_TABLE
+        OmKeyInfo omKeyInfo = omMetadataManager.getKeyTable(BucketLayout.LEGACY)
             .getSkipCache(containerKeyPrefix.getKeyPrefix());
+        if (omKeyInfo == null) {
+          omKeyInfo =
+              omMetadataManager.getKeyTable(BucketLayout.FILE_SYSTEM_OPTIMIZED)
+                  .getSkipCache(containerKeyPrefix.getKeyPrefix());
```

Comment:
Are we not supporting OBS buckets ? Have we tested with OBS buckets ?

Reply:
Yes we do support OBS buckets also :- https://github.com/apache/ozone/pull/4182#issuecomment-1439041949

Reply:
We do support OBS bucket for this also, I have manually tested the changes out as well!
<img width="503" alt="image" src="[https://user-images.githubusercontent.com/98023601/224228270-fbc3ca5b-f2ee-4336-970a-0e73abbafd02.png](https://user-images.githubusercontent.com/98023601/224228270-fbc3ca5b-f2ee-4336-970a-0e73abbafd02.png)">